charmorigin_test.go

```diff
@@ -138,3 +138,40 @@ func (s *CharmOriginSerializationSuite) TestV1ParsingReturnsLatest(c *gc.C) {
 	originResult := s.exportImportVersion(c, originV1, 1)
 	c.Assert(*originResult, jc.DeepEquals, originLatest)
 }
+
+func (s *CharmOriginSerializationSuite) TestPlatformFromSeriesErrorsOnEmpty(c *gc.C) {
+	rval, err := platformFromSeries("")
+	c.Assert(err, gc.NotNil)
+	c.Assert(rval, gc.Equals, "")
+	c.Assert(err.Error(), gc.Equals, "cannot convert empty series to a platform")
+}
+
+func (s *CharmOriginSerializationSuite) TestPlatformFromSeries(c *gc.C) {
+	tests := []struct {
+		expectedVal string
+		series      string
+	}{
+		{
+			expectedVal: "amd64/Ubuntu/20.04",
```

Comment:
I think Ubuntu and Windows etc are supposed to be lowercase.

Reply:
This is what the OS package give us. Should we to lower them ?

Reply:
yes, os package gives us title case but we use lower case